src/main/Main.hs

```diff
@@ -845,6 +853,21 @@ execCmd ExecOpts {..} go@GlobalOpts{..} =
       getPkgOpts wc pkgs =
           map ("-package-id=" ++) <$> mapM (getPkgId wc) pkgs
 
+      getRunCmd args = do
+          pkgComponents <- liftM (map lpvComponents . Map.elems . lpProject) getLocalPackages
+          let executables = filter isCExe $ concatMap Set.toList pkgComponents
+          let argExe = if not (null args) then find (\x -> x == (CExe $ T.pack $ head args)) executables
+                                          else Nothing
+          let firstExe = if not (null executables) then Just $ head executables else Nothing
+          let (exe, args') = if isJust argExe then (argExe, tail args) else (firstExe, args)
```

Comment:
And this could be combined with the pattern match in the above `let` binding to avoid the partial `tail` call.